sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExtractValue.scala

```diff
@@ -195,6 +205,11 @@ case class GetArrayItem(child: Expression, ordinal: Expression)
       baseValue(index)
     }
   }
+
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    defineCodeGen(ctx, ev,
+      (eval1, eval2) => s"(${ctx.javaType(dataType)})$eval1.apply((int)$eval2)")
```

Comment:
cc @davies , in java we can assign `Integer` to `int` without cast, so boxed `T` is not a problem. The failure is:
```java
scala.collection.Seq<org.apache.spark.unsafe.types.UTF8String> primitive3 = null;
...
org.apache.spark.unsafe.types.UTF8String primitive1 = null;
...
primitive1 = primitive3.apply((int)1);

ERROR: Assignment conversion not possible from type "java.lang.Object" to type "org.apache.spark.unsafe.types.UTF8String
```
Does Janino support java generics when compile?